scripts/features.sh

```diff
@@ -31,33 +31,36 @@ sanitize() {
 	echo -n "$v"
 }
 
-vars=
+vars=()
 
 for feature in $1; do
-	if [ "$(type -t gluon_feature_nodefault_${feature})" != 'function' ]; then
+	if [ "$(type -t gluon_feature_nodefault_"${feature}")" != 'function' ]; then
 		echo "gluon-${feature}"
 	fi
 
-	vars="$vars $(sanitize "$feature")=1"
+	vars+=("$(sanitize "$feature")=1")
 done
 
 
 nodefault() {
 	:
 }
 
+# shellcheck disable=SC2086
 packages() {
 	local cond="$(sanitize "$1")"
 	shift
 
 	# We only allow variable names, parentheses and the operators: & | !
-	if [ "$(expr match "$cond" '.*[^A-Za-z0-9_()&|! ].*')" -gt 0 ]; then
+	if grep -q '[^A-Za-z0-9_()&|! ]' <<< "$cond"; then
 		exit 1
 	fi
 
 	# Let will return false when the result of the passed expression is 0,
 	# so we always add 1. This way false is only returned for syntax errors.
-	local ret="$(env -i $vars bash --norc -ec "let _result_='1+($cond)'; echo -n \"\$_result_\"" 2>/dev/null)"
+	set -A
```

Comment:
`set -A` doesn't exist, what is this supposed to do? Please at least build test your changes!

Reply:
Ah sorry that was ksh related. I added it locally for test reasons and accidentally committed it.